src/App.tsx

```diff
@@ -16,46 +48,85 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [reversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [changed, setChanged] = useState(false);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            { 'is-light': sortType !== SortType.ALPHABET },
+          )}
+          onClick={() => {
+            setSortType(SortType.ALPHABET); setChanged(true);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button',
+            'is-success',
+            { 'is-light': sortType !== SortType.LENGTH },
+          )}
+          onClick={() => {
+            setSortType(SortType.LENGTH); setChanged(true);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button',
+            'is-warning',
+            { 'is-light': !reversed },
+          )}
+          onClick={() => {
+            setReversed(current => !current);
+            setChanged(current => !current);
+          }}
```

Comment:
https://www.loom.com/share/696528b61fa84138967a76eba8817567
are you sure about affecting the reset button in this way?